packages/flutter/lib/src/widgets/app.dart

```diff
@@ -234,6 +234,15 @@ typedef PageRouteFactory = PageRoute<T> Function<T>(RouteSettings settings, Widg
 /// Creates a series of one or more initial routes.
 typedef InitialRouteListFactory = List<Route<dynamic>> Function(String initialRoute);
 
+/// Disposes navigation related singletons.
```

Comment:
This will need better documentation. One question that the doc should answer is IMO: Do I as an app developer need to care about disposing singletons? When should I call this method, if ever?